### DISCUSSION TITLE
how to change fzf keybinding when history is enabled?

### DISCUSSION THREAD

**Discussion by niamleeson** [ibhagwan/fzf-lua#1147] [General]
I read the wiki and it says you can change the keybinding in keymap.fzf. Where is that file? Is it possible to remap inside the configuration options?

What I'd like to do is remap ctrl-n/p to something else so I can keep using those keys for going up/down the file list

**Comment by ibhagwan** [comment]
Call `require'fzf-lua'.setup({ keymap = … })`

**Reply by niamleeson** [reply]
Thank you for the response, but which key is that? I didn't see it in the docs

**Reply by ibhagwan** [reply]
From `man fzf`:
```
--history=HISTORY_FILE
Load search history from the specified file and update the file on completion.
When enabled, CTRL-N and CTRL-P are automatically remapped to next-history and prev-history.
```

The mapping happens automatically by fzf when history is enabled, if you wanna remap it use `keymap = { fzf = ['ctrl-p'] = …` to override.

**Reply by niamleeson** [reply]
Sorry for not being clear. When you enable history, ctrl-n and ctrl-p get mapped to previous and next search history. I want to remap those to say, ctrl-f/b and keep using it for going up and down the file list.

How would I remap prev and next search history?

**Reply by ibhagwan** [reply]
I just gave you the answer, look at the available actions in `man fzf` and use `keymap.fzf` to remap these to whatever you want.

**Reply by niamleeson** [reply]
What is keymap.fzf?

**Reply by ibhagwan** [reply]
This
https://github.com/ibhagwan/fzf-lua/discussions/1147#discussioncomment-9205431

**Reply by niamleeson** [reply]
Okay thank you

**Reply by ibhagwan** [reply]
This should do what you want, if it doesn’t consult `man fzf`:
```lua
require'fzf-lua'.setup({
  keymap = {
    fzf = {
      ['ctrl-f'] = 'next-history',
      ['ctrl-b'] = 'prev-history',
      ['ctrl-n'] = 'down',
      ['ctrl-p'] = 'up',
    }
  }
})
```
